File: android/src/main/kotlin/com/syncrhrohealth/bandpass_ble_connection/plugin/devicecore/enum/DeviceCoreEvent.kt
package com.syncrhrohealth.bandpass_ble_connection.plugin.devicecore.enum

/** Make sure the event value matches the DeviceCoreEvent on Flutter side */
enum class DeviceCoreEvent(val value: String) {
    // Connection
    ON_DEVICE_FOUND("onDeviceFound"),
    ON_DEVICE_CONNECTED("onDeviceConnected"),
    ON_DEVICE_DISCONNECTED("onDeviceDisconnected"),

    // Data response
    ON_DEVICE_INFO_RSP("onDeviceInfoRsp"),
    ON_IMU_DATA_RSP("onImuDataRsp"),
}

File: ios/Classes/Plugin/DeviceCore/Enum/DeviceCoreEvent.swift
//
//  DeviceCoreEvent.swift
//  Runner
//
//  Created by MAC on 12/3/25.
//

import Foundation

/// Make sure the event value matches the `DeviceCoreEvent` on the Flutter side
enum DeviceCoreEvent: String {
    // Connection
    case onDeviceFound = "onDeviceFound"
    case onDeviceConnected = "onDeviceConnected"
    case onDeviceDisconnected = "onDeviceDisconnected"

    // Data response
    case onDeviceInfoRsp = "onDeviceInfoRsp"
    case onImuDataRsp = "onImuDataRsp"
    case onBatteryLevelRsp = "onBatteryLevelRsp"
}
